core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ExpressionEquivalence.java

```diff
@@ -262,9 +263,7 @@ public int compare(RowExpression left, RowExpression right)
                     }
                 }
 
-                // value is some random type (say regex), so we just randomly choose a greater value
-                // todo: support all known type
-                return -1;
+                throw new IllegalArgumentException("Cannot compare " + leftConstant + " and " + rightConstant + " of type " + javaType);
```

Comment:
left java type and right java type may be different

Reply:
Yeah. It seems that I need to implement Block sorting here as well.